x/bsonx/bsoncore/bsoncore.go

```diff
@@ -335,27 +335,27 @@ func ReadBinary(src []byte) (subtype byte, bin []byte, rem []byte, ok bool) {
 // AppendUndefinedElement will append a BSON undefined element using key to dst
 // and return the extended buffer.
 func AppendUndefinedElement(dst []byte, key string) []byte {
-	return AppendHeader(dst, bsontype.Undefined, key)
+	return AppendHeader(dst, TypeUndefined, key)
 }
 
 // AppendObjectID will append oid to dst and return the extended buffer.
-func AppendObjectID(dst []byte, oid primitive.ObjectID) []byte { return append(dst, oid[:]...) }
+func AppendObjectID(dst []byte, oid [idLen]byte) []byte { return append(dst, oid[:]...) }
```

Comment:
Instead of defining `idLen = 12` and then using type `[idLen]byte` everywhere, you can use a type alias to prevent having to type `[12]byte` everywhere.

E.g.
```go
type objectID = [12]byte

func AppendObjectID(dst []byte, oid objectID) []byte { return append(dst, oid[:]...) }
```

To callers outside of the `bsoncore` package, the function signature will just look like
```go
func AppendObjectID(dst []byte, oid [12]byte) []byte
```
but you won't have to type `[12]byte` everywhere.